boards/pic32-wifire/wifire.c

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2016, Imagination Technologies Limited and/or its
+ *                 affiliated group companies.
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+#include "bitarithm.h"
+#include "periph/uart.h"
+#include "periph_uart.h"
+
+static inline void w32(uint32_t addr, uint32_t val)
+{
+    *((volatile uint32_t *)addr) = val;
+}
+
+void board_init(void)
+{
+    /*
+     * Setup pin mux for UART4 this is the one connected
+     * to the ftdi chip (usb<->uart)
+     */
+    w32(U4RXR, 0xb);            /* connect pin RPF2 to UART 4 RX */
```

Comment:
I agree a whole pin-mux driver needs implemented and we have started work on one.
peripheral support is a bit hacky at the moment, we just wanted to get the UART going so we can show that the 'core' architecture port works. get that merged then improve pic32 device support so it can be used for actual products.